modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java

```diff
@@ -157,6 +163,19 @@
      */
     private final boolean isPersistenceEnabled;
 
+    /**
+     * Task statuses update monitors.
+     * Guarded by {@link #lock}.
```

Comment:
I hope that this lock is really what we needed!

Reply:
We discussed it in a slack - yes, it is needed and used wherever necessary: adding / removing a task status monitor, updating task statuses.